### PR TITLE
Copr: Vectorize `LogicalXor`

### PR DESCRIPTION
**Pull request by koushiro**
PCP #5751

Signed-off-by: koushiro <koushiro.cqx@gmail.com>

###  What have you changed?

Vectorize `LogicalXor`

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

File: components/tidb_query/src/rpn_expr/impl_op.rs [hunk 1]
```diff
@@ -27,6 +27,18 @@ pub fn logical_or(arg0: &Option<i64>, arg1: &Option<i64>) -> Result<Option<i64>>
     })
 }
 
+#[rpn_fn]
+#[inline]
+pub fn logical_xor(arg0: &Option<i64>, arg1: &Option<i64>) -> Result<Option<i64>> {
+    // evaluates to 1 if an odd number of operands is nonzero, otherwise 0 is returned.
+    Ok(match (arg0, arg1) {
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_op.rs]: Prefer to use the same implementation as the non-vectorized one, which is:

```
Ok(Some(((arg0 == 0) ^ (arg1 == 0)) as i64))
```

Comment (koushiro) [reply components/tidb_query/src/rpn_expr/impl_op.rs]: Okay :)

**Pull request merged**